summer/stratification.py

```diff
@@ -129,21 +133,30 @@ def add_flow_adjustments(
 
     def get_flow_adjustment(self, flow) -> dict:
         """
-        Returns the most recently added flow adjustment that matches a given flow.
+        Note that the loop structure implies that if the user has requested multiple adjustments that apply to a single
+        combination of strata (across multiple stratifications), then only the last one that is applicable will be used
+        - because the last request will over-write the earlier ones in the loop.
+
         """
         flow_adjustments = self.flow_adjustments.get(flow.name, [])
         matching_adjustment = None
+
+        # Loop over all the requested adjustments.
         for adjustment, source_strata, dest_strata in flow_adjustments:
-            # Skip any adjustments that don't match the flow's source compartment strata.
-            msg = f"Source strata specified in flow adjustment of {self.name} but {flow.name} does not have a source"
+
+            # Skip any adjustments that don't match the flow's source and dest compartment strata.
+            # For entry flows:
+            msg = f"Source strata requested in flow adjustment of {self.name}, but {flow.name} does not have a source"
             assert not (source_strata and not flow.source), msg
-            if source_strata and flow.source and not flow.source.has_strata(source_strata):
-                continue
 
-            # Skip any adjustments that don't match the flow's dest compartment strata.
-            msg = f"Dest strata specified in flow adjustment of {self.name} but {flow.name} does not have a dest"
+            # For exit flows:
+            msg = f"Dest strata requested in flow adjustment of {self.name}, but {flow.name} does not have a dest"
             assert not (dest_strata and not flow.dest), msg
-            if dest_strata and flow.dest and not flow.dest.has_strata(dest_strata):
+
```

Comment:
Using \ to denote newlines is generally considered bad Python style
I recommend breaking this up into more intermediate steps

```
is_source_no_match = source_strata and flow.source and not flow.source.has_strata(source_strata)
is_dest_no_match = (dest_strata and flow.dest and not flow.dest.has_strata(dest_strata))
if is_source_no_match or is_dest_no_match:
    continue
```